Socket timeout issue since Java update (unable to connect to local server)

I'm not sure why this is happening but ever since I updated my Java packages, I'm unable to connect to my local server for testing.  I'm on Arch Linux and the update was from openjdk 8.u292-1 (working) to openjdk 8 .u312-2 (not working). 

Downgrading the packages resolves the issue but I'm a little hesitant to do that as a permanent solution (log4j security issues for one...)

From the logs, the server seems to start up fine but I get a socket timeout error in the client logs.

```
INFO  2022-01-28 08:45:23,565 Connect: connecting to server localhost:17171                                              =>[SwingWorker-pool-5-thread-4] SessionImpl$5.work 
WARN  2022-01-28 08:45:53,712 Cannot connect                                                                             =>[SwingWorker-pool-5-thread-3] SessionImpl.handleCannotConnectException 
org.jboss.remoting.CannotConnectException: Error setting up client lease upon performing connect.
	at org.jboss.remoting.Client.connect(Client.java:1928)
	at org.jboss.remoting.Client.connect(Client.java:737)
	at mage.remote.SessionImpl$5.work(SessionImpl.java:429)
	at mage.remote.SessionImpl$RemotingTask$1.doInBackground(SessionImpl.java:95)
	at mage.remote.SessionImpl$RemotingTask$1.doInBackground(SessionImpl.java:91)
	at javax.swing.SwingWorker$1.call(SwingWorker.java:295)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at javax.swing.SwingWorker.run(SwingWorker.java:334)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.Exception: Error setting up client lease
	at org.jboss.remoting.MicroRemoteClientInvoker.establishLease(MicroRemoteClientInvoker.java:563)
	at org.jboss.remoting.Client.setupClientLease(Client.java:2070)
	at org.jboss.remoting.Client.connect(Client.java:1924)
	... 10 more
Caused by: org.jboss.remoting.InvocationFailureException: Socket timed out.  Waited 29972 milliseconds for response while calling on InvokerLocator [bisocket://192.168.21.246:17171/?onewayThreadPool=mage.remote.CustomThreadPool&serializationtype=jboss]; nested exception is: 
	java.net.SocketTimeoutException: Read timed out
	at org.jboss.remoting.transport.socket.SocketClientInvoker.handleException(SocketClientInvoker.java:137)
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.handleOtherException(MicroSocketClientInvoker.java:1104)
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.transport(MicroSocketClientInvoker.java:966)
	at org.jboss.remoting.transport.bisocket.BisocketClientInvoker.transport(BisocketClientInvoker.java:470)
	at org.jboss.remoting.MicroRemoteClientInvoker.invoke(MicroRemoteClientInvoker.java:169)
	at org.jboss.remoting.MicroRemoteClientInvoker.establishLease(MicroRemoteClientInvoker.java:529)
	... 12 more
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:265)
	at java.io.FilterInputStream.read(FilterInputStream.java:83)
	at org.jboss.serial.io.JBossObjectInputStream.read(JBossObjectInputStream.java:193)
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.readVersion(MicroSocketClientInvoker.java:1366)
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.transport(MicroSocketClientInvoker.java:920)
	... 15 more
INFO  2022-01-28 08:45:53,736 Disconnecting DONE                                                                         =>[SwingWorker-pool-5-thread-3] SessionImpl.disconnect 
INFO  2022-01-28 08:45:53,738 Disconnected from user mode                                                                =>[SwingWorker-pool-5-thread-3] MageFrame.disconnected 
```

--- a/Mage.Common/src/main/java/mage/remote/Connection.java
+++ b/Mage.Common/src/main/java/mage/remote/Connection.java
@@ -40,7 +40,7 @@ public class Connection {
 //    private boolean confirmEmptyManaPool;
 //    private String flagName;
 //    private UserSkipPrioritySteps userSkipPrioritySteps;
-    private static final String serialization = "?serializationtype=jboss";
+    private static final String serialization = "?serializationtype=java";
     private static final String transport = "bisocket";
     private static final String threadpool = "onewayThreadPool=mage.remote.CustomThreadPool";
 

--- a/Mage.Common/src/test/java/mage/remote/ConnectionTest.java
+++ b/Mage.Common/src/test/java/mage/remote/ConnectionTest.java
@@ -72,7 +72,7 @@ public class ConnectionTest {
         void serialisation() {
             final String query = make(testeeBuilder).getQuery();
 
-            assertThat(query).contains("serializationtype=jboss");
+            assertThat(query).contains("serializationtype=java");
         }
 
         @Test

